code/modules/clothing/suits/armor.dm

```diff
@@ -55,14 +55,52 @@
 	slowdown = LIGHT_SLOWDOWN
 	style_coverage = COVERS_TORSO|COVERS_UPPER_ARMS|COVERS_UPPER_LEGS
 
-/obj/item/clothing/suit/armor/vest/full/security
+
+//This has specifically been bodged so that I can give the flak vest toggle-able buttons -VaNdU Jr
+/obj/item/clothing/suit/armor/vest/toggle/security/full
 	name = "full security armor"
 	desc = "A tactical armor vest, but with shoulderpads and knee pads included to cover all parts of the body. Not designed for serious operations."
 	icon_state = "armor_security_fullbody"
+	icon_open = "armor_security_fullbody_open"
+	icon_closed = "armor_security_fullbody"
+	blood_overlay_type = "armor"
+	slowdown = 0.1
+	body_parts_covered = UPPER_TORSO|LOWER_TORSO|ARMS|LEGS // kneepads and shoulderpads, so it covers arms and legs
+	matter = list(
+		MATERIAL_STEEL = 10, // contains a lil bit more steel because of arm+leg prot
+		MATERIAL_PLASTEEL = 1
+	)
+	slowdown = LIGHT_SLOWDOWN
+	style_coverage = COVERS_TORSO|COVERS_UPPER_ARMS|COVERS_UPPER_LEGS
 
-/obj/item/clothing/suit/armor/vest/security
+/obj/item/clothing/suit/armor/vest/toggle/security
 	name = "security armor"
 	icon_state = "armor_security"
+	icon_open = "armor_security_open"
+	icon_closed = "armor_security"
+
+/obj/item/clothing/suit/armor/vest/toggle
+	bad_type = /obj/item/clothing/suit/armor/vest/toggle
+	/obj/item/clothing/suit/armor/vest/toggle/var/icon_open
+	/obj/item/clothing/suit/armor/vest/toggle/var/icon_closed
+	/obj/item/clothing/suit/armor/vest/toggle/verb/toggle()
+		set name = "Toggle Vest Buttons"
+		set category = "Object"
+		set src in usr
+		if(!usr.canmove || usr.stat || usr.restrained())
+			return 0
+
+		if(icon_state == icon_open) //Will check whether icon state is currently set to the "open" or "closed" state and switch it around with a message to the user
+			icon_state = icon_closed
+			to_chat(usr, "You button up the vest.")
+		else if(icon_state == icon_closed)
+			icon_state = icon_open
+			to_chat(usr, "You unbutton the vest.")
+		else //in case some goofy admin switches icon states around without switching the icon_open or icon_closed
+			to_chat(usr, "You attempt to button-up the velcro on your [src], before promptly realising how silly you are.")
+			return
+		update_wear_icon()	//so our overlays update
+
```

Comment:
```suggestion
	var/icon_open = "armor_security_open"
	var/icon_closed = "armor_security"

/obj/item/clothing/suit/armor/vest/toggle/verb/toggle()
	set name = "Toggle Vest Buttons"
	set category = "Object"
	set src in usr
	if(!usr.canmove || usr.stat || usr.restrained())
		return

	if(icon_state == icon_open) //Will check whether icon state is currently set to the "open" or "closed" state and switch it around with a message to the user
		icon_state = icon_closed
		to_chat(usr, "You button up the vest.")
	else if(icon_state == icon_closed)
		icon_state = icon_open
		to_chat(usr, "You unbutton the vest.")
	else //in case some goofy admin switches icon states around without switching the icon_open or icon_closed
		to_chat(usr, "You attempt to button-up the velcro on your [src], before promptly realising how silly you are.")
		return
	update_wear_icon()	//so our overlays update

```